jujugui/static/gui/src/app/components/login/login.js

```diff
@@ -98,6 +97,17 @@ YUI.add('login-component', function() {
     },
 
     render: function() {
+      var buttons = [];
+      var MacaroonButton = this._generateLoginWithMacaroonButton();
+      if (MacaroonButton) {
+        buttons.push(MacaroonButton);
+      }
+      buttons.push({
```

Comment:
I think this button should be the first one. When a user tabs from username to password then the next tab will be to login not login with SSO.